pkg/plugin/resources.go

```diff
@@ -211,10 +211,13 @@ func (app *App) handleVectorSearch(w http.ResponseWriter, req *http.Request) {
 
 // registerRoutes takes a *http.ServeMux and registers some HTTP handlers.
 func (a *App) registerRoutes(mux *http.ServeMux, settings Settings) {
-	if settings.OpenAI.UseAzure {
-		mux.Handle("/openai/", newAzureOpenAIProxy(settings))
-	} else {
+	switch settings.OpenAI.Provider {
+	case openAIProviderOpenAI:
 		mux.Handle("/openai/", newOpenAIProxy(settings))
+	case openAIProviderAzure:
+		mux.Handle("/openai/", newAzureOpenAIProxy(settings))
+	default:
+		log.DefaultLogger.Warn("Unknown OpenAI provider configured", "provider", settings.OpenAI.Provider)
```

Comment:
If the provider isn't valid, this quietly logs a warning but then (I think) leaves the plugin in a broken state.

How will it fail for the developer or user who ends up here? Is there a way to make the misconfiguration more visible?

--- Oh, I see below we're forcing a provider to be chosen. So this is dead code unless there's a bug. If there's a bug, what should happen?

Reply:
I did look for a way to handle this in the plugin but from what I can see there isn't a good way 🤔 I don't think plugin backends have the ability to validate the JSON data when it's posted to Grafana (or provided in a provisioning file); the best we could do is exit very early, but even then the logs are pretty cryptic 😞